etl-pipeline/processes/ingest/doab.py

```diff
@@ -3,86 +3,46 @@
 from logger import create_log
 from mappings.doab import DOABMapping
 from managers import DBManager, DOABLinkManager, S3Manager, RabbitMQManager
-from model import get_file_message
+from model import get_file_message, Source
 from ..record_buffer import RecordBuffer, Record
 from .. import utils
 
+from logger import create_log
+from model import Source
+from services import DSpaceService
+from ..record_ingestor import RecordIngestor
+from .. import utils
+
 logger = create_log(__name__)
 
+
 class DOABProcess():
     DOAB_BASE_URL = 'https://directory.doabooks.org/oai/request?'
     DOAB_IDENTIFIER = 'oai:directory.doabooks.org'
 
     def __init__(self, *args):
-        self.params = utils.parse_process_args(*args)
-
-        self.db_manager = DBManager()
-        self.db_manager.create_session()
-
-        self.record_buffer = RecordBuffer(self.db_manager)
-
-        self.s3_manager = S3Manager()
-        self.s3_bucket = os.environ['FILE_BUCKET']
-
-        self.file_queue = os.environ['FILE_QUEUE']
-        self.file_route = os.environ['FILE_ROUTING_KEY']
-
-        self.rabbitmq_manager = RabbitMQManager()
-        self.rabbitmq_manager.create_connection()
-        self.rabbitmq_manager.create_or_connect_queue(self.file_queue, self.file_route)
-
         self.dspace_service = DSpaceService(base_url=self.DOAB_BASE_URL, source_mapping=DOABMapping)
-
-    def runProcess(self):
-        try:
-            if self.params.record_id is not None:
-                record = self.dspace_service.get_single_record(record_id=self.params.record_id, source_identifier=self.DOAB_IDENTIFIER)
-                
-                self._process_record(record)
-                self.record_buffer.flush()
-                self._log_results()
-                
-                return
-
-            records = self.dspace_service.get_records(
-                start_timestamp=utils.get_start_datetime(process_type=self.params.process_type, ingest_period=self.params.ingest_period),
-                offset=self.params.offset,
-                limit=self.params.limit
+        self.params = utils.parse_process_args(*args)
+        self.record_ingestor = RecordIngestor(source=Source.DOAB.value)
+
+    def runProcess(self) -> int:
+        start_timestamp = utils.get_start_datetime(
+            process_type=self.params.process_type,
+            ingest_period=self.params.ingest_period,
+        )
+        
+        if self.params.record_id is not None:
```

Comment:
I think similar to muse we can have a get_record function in the source_service which is implemented in the child classes

Reply:
I originally did put a get_record function in the source_service and then I realized that we would have to implement the function in every child classes despite if they implemented it or not. I could have some of the child classes just have the method and only have pass in the code block for now until we decide at a later time to implement each child source service with logic to retrieve a single record. What are your thoughts?